make/common/JdkNativeCompilation.gmk

```diff
@@ -189,6 +189,24 @@ define SetupJdkLibraryBody
     $1_EXTRA_HEADER_FLAGS := $$(addprefix -I, $$($1_PROCESSED_EXTRA_HEADER_DIRS))
   endif
 
+  ifneq ($$($1_DEFAULT_CFLAGS), false)
+    # Set the default flags first to be able to override
+    $1_CFLAGS := $$(CFLAGS_JDKLIB) $$($1_CFLAGS)
+    ifneq ($$($1_CFLAGS_FILTER_OUT), )
+      $$(foreach flag, $$($1_CFLAGS_FILTER_OUT), \
+        $$(eval $1_CFLAGS := $(filter-out $$(flag), $$($1_CFLAGS)) \
+      ))
+    endif
+
+    # Set the default flags first to be able to override
+    $1_CXXFLAGS := $$(CXXFLAGS_JDKLIB) $$($1_CXXFLAGS)
+    ifneq ($$($1_CXXFLAGS_FILTER_OUT), )
+      $$(foreach flag, $$($1_CXXFLAGS_FILTER_OUT), \
+        $$(eval $1_CXXFLAGS := $(filter-out $$(flag), $$($1_CXXFLAGS)) \
+      ))
```

Comment:
Ah, I did not realize filter-out took a list of words to remove. So `$(filter-out foo bar, bar foo)` would return the empty string, and I thought it would return `bar foo`. That indeed simplifies things, thank you!